src/lib/expand-unit.ts

```diff
@@ -49,12 +57,41 @@ export const expandCrossbellNote = async ({
     let rendered
     if (expandedNote.metadata?.content?.content) {
       const { renderPageContent } = await import("~/markdown")
-      rendered = renderPageContent(expandedNote.metadata.content.content, true)
+
+      const processedContent = removeMarkdown(
+        expandedNote.metadata.content.content.replace(/```[^]+?```/g, ""),
+      )
+      const detectedLang = detectLanguage(processedContent)
+      expandedNote.metadata.content.originalLanguage = detectedLang
+
+      if (translateTo) {
+        try {
+          const translation = await (
+            await fetch(
+              `${SITE_URL}/api/translate-note?` +
+                new URLSearchParams({
+                  cid: toCid(expandedNote.metadata.uri || ""),
+                  fromLang: expandedNote.metadata.content.originalLanguage,
+                  toLang: translateTo,
+                } as any),
+            )
+          ).json()
+
+          if (translation.data) {
+            expandedNote.metadata.content.content = translation.data.content
+            expandedNote.metadata.content.title = translation.data.title
+          }
+        } catch (e) {
+          // do nothing
+        }
+      }
+
+      rendered = renderPageContent(expandedNote.metadata.content.content!, true)
       if (keyword) {
-        const position = expandedNote.metadata.content.content
-          .toLowerCase()
+        const position = expandedNote.metadata.content
+          .content!.toLowerCase()
```

Comment:
Because `expandedNote.metadata.content. content` has been reset to the line of 81.  But TS doesn't know whether new content of `translation.data` has a valid value.
```
if (translation.data) {
    expandedNote.metadata.content.content = translation.data.content
    expandedNote.metadata.content.title = translation.data.title
}
```
Now, I have modified it to confirm the new value is valid.